Add some debugging to try and get more info about a segfault we're seeing
through some asserts.

Upstream-Status: Inappropriate [debug for segfault]

Index: qemu-2.4.0/hw/net/smc91c111.c
===================================================================
--- qemu-2.4.0.orig/hw/net/smc91c111.c
+++ qemu-2.4.0/hw/net/smc91c111.c
@@ -157,6 +157,7 @@ static void smc91c111_pop_rx_fifo(smc91c
     int i;
 
     s->rx_fifo_len--;
+    assert(s->rx_fifo_len < NUM_PACKETS);
     if (s->rx_fifo_len) {
         for (i = 0; i < s->rx_fifo_len; i++)
             s->rx_fifo[i] = s->rx_fifo[i + 1];
@@ -174,6 +175,7 @@ static void smc91c111_pop_tx_fifo_done(s
 
     if (s->tx_fifo_done_len == 0)
         return;
+    assert(s->tx_fifo_done_len <= NUM_PACKETS);
     s->tx_fifo_done_len--;
     for (i = 0; i < s->tx_fifo_done_len; i++)
         s->tx_fifo_done[i] = s->tx_fifo_done[i + 1];
@@ -201,6 +203,7 @@ static void smc91c111_do_tx(smc91c111_st
         return;
     if (s->tx_fifo_len == 0)
         return;
+    assert(s->tx_fifo_len <= NUM_PACKETS);
     for (i = 0; i < s->tx_fifo_len; i++) {
         packetnum = s->tx_fifo[i];
         p = &s->data[packetnum][0];
@@ -674,6 +677,8 @@ static ssize_t smc91c111_receive(NetClie
     packetnum = smc91c111_allocate_packet(s);
     if (packetnum == 0x80)
         return -1;
+
+    assert(s->rx_fifo_len < NUM_PACKETS);
     s->rx_fifo[s->rx_fifo_len++] = packetnum;
 
     p = &s->data[packetnum][0];
